protocols/pcr_prep/pcr_prep.py

```diff
@@ -0,0 +1,868 @@
+                   "target_labware",
+                   "mastermix_labware",
+                   "DNA_sample_plate",
+                   "mod_temperature",
+                   "master_mix_csv",
+                   "mastermix_volume",
+                   "DNA_volume"
+                   )
+
+    # Do input error checking here
+    if (not _left_pipette_part_1 and not _right_pipette_part_1 and
+            _create_mastermix):
+        raise Exception('You have to select at least 1 pipette for part 1')
+
+    if (not _left_pipette_part_2 and not _right_pipette_part_2
+            and not _use_same_pipettes):
+        raise Exception('You have to select at least 1 pipette for part 2')
+
+    # Error check the CSV file
+    # 1. Check that the total volume of mastermix that we create is enough
+    # to pipette into each target well
+    # Fields: Reagent | slot | well | volume
+
+    mastermix_matrix = [
+        [cell.strip() for cell in line.split(',')]
+        for line in _master_mix_csv.splitlines()[1:] if line
+    ]
+
+    mastermix_volume_sum = 0
+    for row in mastermix_matrix:
+        mastermix_volume_sum += int(row[3])
+
+    if mastermix_volume_sum < _mastermix_volume * _number_of_samples:
+        raise Exception("The mastermix that you are creating" +
+                        "does not have sufficient volume for your DNA sampes")
+
+    # Error check that there's auxiliary labware if the CSV is requesting it
+    if _aux_pcr_reagent_labware is None:
+        for row in mastermix_matrix:
+            if row[1] == 2:
+                raise Exception("The CSV mastermix file makes use of aux. " +
+                                "labware, but none has been selected")
+    # define all custom variables above here with descriptions:
+
+    # load modules
+
+    '''
+
+    Add your modules here with:
+
+    module_name = ctx.load_module('{module_loadname}', '{slot number}')
+
+    Note: if you are loading a thermocycler, you do not need to specify
+    a slot number - thermocyclers will always occupy slots 7, 8, 10, and 11.
+
+    For all other modules, you can load them on slots 1, 3, 4, 6, 7, 9, 10.
+
+    '''
+
+    # Loading temperature modules here
+    # temp_mod_list = [None, None, None]
+    """
+    Slot 1: In part 1 this will hold the mastermix target container, or the
+    DNA/mastermix plate (MDNAP)
+    Slot 4: Part 1: PCR components, Part 2: DNA samples
+    Slot 7: Part 1: Auxiliary PCR components, Part 2: Mastermix container
+    """
+    """temp_mod_slots = ['1', '4', '7']
+    temp_mod_type_list = [_temp_mod_slot_1, _temp_mod_slot_4,
+                          _temp_mod_slot_7]
+    for i, slot, type in zip([0, 1, 2],
+                             temp_mod_slots,
+                             temp_mod_type_list):
+        if type is not None:
+            temp_mod_list[i] = ctx.load_module(type, slot)"""
+    temp_mod_slot_4, temp_mod_slot_7, temp_mod_slot_10 = None, None, None
+
+    def load_temp_mods():
```

Comment:
Interesting approach. I think this could be accomplished with a for loop instead of a standalone function.
`for slot in [temp_mod_slot_4,...]:
    if slot is not None:
        var = ctx.load_module(...`